### ISSUE TITLE
The general 'wire-up' to the data issue ... 

### ISSUE THREAD

**Issue by ddkapan** [calacademy-research/COMB#26]
Need to figure out some best practices && then implement them to wire our code to the data
- [ ] Best practices, new tools, GoogleCloudPlatform?
- [ ] [SpatialPtCtPlots.R](https://github.com/calacademy-research/COMB/blob/main/spatial_plots/src/SpatialPtCtPlots.R)

**Comment by matth79** [comment]
Service account as a way of accessing Google Drive is one alternative to consider.

**Comment by ddkapan** [comment]
Thanks @matth79 does a service account cost $?

**Comment by matth79** [comment]
I believe a service account does not cost money by creating it in a cloud project for which billing is disabled, though I'm not 100% sure of that.

However, now that you got person user accounts working for everyone, sharing around a common credentials file doesn't seem necessary or best. It also was even less easy to understand than the user OAuth token method.

If we end up with automated processes that need to access Drive, service accounts would be the right tool for that.

I did create one and verify it works for our scripts by adding a parameter to drive_auth.

```
drive_auth(path='service.json')
```

**Comment by matth79** [comment]
Specific cases include

#54 #23

**Comment by matth79** [comment]
I tried using symlinks for data dependencies in [this commit](https://github.com/calacademy-research/COMB/pull/68/commits/7cd9082ea323952f2a64b758a3da4d98d90443bd).

It works, but it made a mess of .gitignore.

If we want the symlinks, maybe it would be good to consider not having lines ending in "input/" in .gitignore except for "original" human-created data like aru2point.csv, latlong.csv, etc. Once you have an ignore ending in a slash, you can't make exceptions underneath it.

A good convention would be that chunk C only ever read input from C/input, and if that input is the output of another chunk B, then the commit that adds the dependent code to C needs to also add a symlink from C/input into B/output, similar to the commit linked above.

If people adhere to that and we don't .gitignore input/, then "git status" will remind them which symlinks they need to add, if they have already created them locally.

**Comment by ddkapan** [comment]
I actually ignored matt and pushed a .gitignore on a branch and merged the pull request.  This is b/c Mark Schulist (elsewhere) confirmed that the symlinks need to be generated especially for each user so synchronizing them with GitHub wasn't going to work anyway.  Correct @mschulist?  What do you think @matt-har-vey?

**Comment by mschulist** [comment]
> Symlinks need to be generated especially for each user so synchronizing them with GitHub wasn't going to work anyway. 
Yes, this is true. We can include a line of code that will (re)create them. We could also just reference the outputs from another script, but that makes it more difficult to see what the inputs are for a given script.

**Comment by ddkapan** [comment]
- [ ] Go over symlink creation code for each occurrence of an /input/ directory in the codebase to finish this
